src/NrFileCompressor.cpp

```diff
@@ -64,10 +75,13 @@ int NrFileCompressor::fileCompress(QString const& i_fileName, NrFileCompressor::
 QString
 NrFileCompressor::getCompressedFilename(const QString &i_fileName, NrFileCompressor::compressedFileFormatEnum i_algo)
 {
-    if (i_algo == NrFileCompressor::GZIP_ARCHIVE) {
-        return i_fileName + GZIP_EXT;
-    } else {
-        return i_fileName + ZIP_EXT;
+    switch(i_algo) {
+        case NrFileCompressor::GZIP_ARCHIVE:
+            return i_fileName + GZIP_EXT;
+        case NrFileCompressor::ZIP_ARCHIVE:
+            return calculateNameCompliantWithZipAlgoMiniZ(i_fileName) + ZIP_EXT;
+        default:
+            return i_fileName;
```

Comment:
Unsupported format. Shouldn't you return an error?

Reply:
you mean the default? that matches the NO_COMPRESSION enum value that should return the unchanged filename